File: Task-03/helloworld.cpp
#include <iostream>
#include <vector>

void printArray(const std::vector<int>& arr) {
    std::cout << "Array: [";
    for (size_t i = 0; i < arr.size(); ++i) {
        std::cout << arr[i];
        if (i < arr.size() - 1) {
            std::cout << ", ";
        }
    }
    std::cout << "]" << std::endl;
}

int binarySearch(const std::vector<int>& arr, int target) {
    int left = 0;
    int right = arr.size() - 1;

    while (left <= right) {
        int mid = left + (right - left) / 2;

        if (arr[mid] == target) {
            return mid;
        } else if (arr[mid] < target) {
            left = mid + 1;
        } else {
            right = mid - 1;
        }
    }

    return -1; // To return -1 if the target is not found
}

int main() {
    std::vector<int> arr = {1, 2, 4, 5, 7, 8, 9, 12, 15, 18, 21};
    int target = 7;

    printArray(arr);
    std::cout << "Target: " << target << std::endl;

    int result = binarySearch(arr, target);

    if (result != -1) {
        std::cout << "Target found at index: " << result << std::endl;
    } 
}

File: Task-03/helloworld.java
public class BinarySearch {
    public static void main(String[] args) {
        int[] array = { 1, 3, 5, 7, 9, 11, 13, 15, 17, 19 };
        int target = 13;

        System.out.print("Array: ");
        printArray(array);

        System.out.println("Target: " + target);

        int index = binarySearch(array, target);
        if (index != -1) {
            System.out.println("Target found at index: " + index);
        } else {
            System.out.println("Target not found in the array.");
        }
    }
    public static int binarySearch(int[] array, int target) {
        int left = 0;
        int right = array.length - 1;

        while (left <= right) {
            int mid = left + (right - left) / 2;
            if (array[mid] == target) {
                return mid;
            }
            if (array[mid] < target) {
                left = mid + 1;
            }
            else {
                right = mid - 1;
            }
        }
        return -1;
    }
    public static void printArray(int[] array) {
        for (int num : array) {
            System.out.print(num + " ");
        }
        System.out.println();
    }
}
